neuronsimulator/views.py

```diff
@@ -1,12 +1,140 @@
+import os
+
+import numpy as np
+import plotly.graph_objects as go
+import wormfunconn as wfc
+from django.conf import settings
 from django.shortcuts import render
-from neuronsimulator.forms import ParamForm
+from neuronsimulator.forms import NeuronInputParamForm, ParamForm
 from neuronsimulator.models import Neuron
+from plotly.offline import plot
 
 
 def home(request):
     return render(request, "home.html")
 
 
+def plot_neural_responses(request):
+    context = {}
+    params = {}
+
+    # all neurons
+    neurons = Neuron.objects.all()
+
+    # form
+    my_form = NeuronInputParamForm(request.POST or None)
+
+    # context
+    context = {"neurons": neurons, "form": my_form}
+
+    if my_form.is_valid():
+        params = my_form.cleaned_data
+        # get input parameters for calling wormfunconn
+        stim_neu_id = params["stim_neu_id"]
+        resp_neu_ids = params["resp_neu_ids"]
+        resp_neu_ids = resp_neu_ids.split(",")
+        nt = int(params["nt"])
+        t_max = float(params["t_max"])
+        stim_type = params["stim_type"]
+        dur = float(params["dur"])
+
+        # print("stim_neu_id", stim_neu_id)
+        # print("resp_neu_ids", resp_neu_ids)
+
+        # Get atlas folder and file name
+        folder = os.path.join(settings.MEDIA_ROOT, "atlas/")
+        # use mock file for now, need get the file from the lab
+        fname = "mock.pickle"
+
+        # Create FunctionalAtlas instance from file
+        if os.path.isfile(os.path.join(folder, fname)):
+            funatlas = wfc.FunctionalAtlas.from_file(folder, fname)
+        else:
+            raise FileNotFoundError("Input Atlas file was not found")
+
+        # calculate dt from t_max and nt
+        dt = t_max / nt
```

Comment:
NB. I know that the form has a min value of 1 in the form validation, but for security concerns (because client-side validation can be circumvented), I think doing a server-side code check that `nt` is greater than 0 would prevent an exception upon invalid input.